### PR TITLE
SQL_DDL_FUNC_HW

### PR DESCRIPTION
**Pull request by olegarslanov**
Now I really know how added files ... only one not forget it

### CODE REVIEW

**Comment by geroget** [comment]
Hello Oleg, very nice work, thank you!

A few general comments:
- lack of schema name before the table name;
- too poor comments. This really lowers your grade! Don't forget to add comments that will let us understand why you chose this particular solution and what you are doing.

Task 1 - 2

how did you check this requirement: categories with at least one sale?

task3 
add a RAISE NOTICE/EXCEPTION about an invalid parameter 

task 5 
impossible to insert a new movie with default languages. ERROR: The language with the identifier “Klingon” does not exist!
you need to add a part to insert a new language.

++ for additional solutions

**Comment by olegarslanov** [comment]
Hallo Tatjana. Please review my new commit in this PR. I hope I understand all like must be 

p.s. Thank You for give me a chance :)

**Comment by geroget** [comment]
the lack of comments still does not allow me to understand why you use such a complex filter in the where clause in tasks 1-2
task 5 
 you don't need to add a separate insert script. you need to include it in the function

File: SQL_DDL_Func_Oleg_Arslanov_HW.sql [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+--Task 5. Create procedure language functions
+--Create a procedure language function called 'new_movie' that takes a movie title as a parameter and inserts a new movie with the given title in the film table.
+--The function should generate a new unique film ID, set the rental rate to 4.99, the rental duration to three days, the replacement cost to 19.99. 
+--The release year and language are optional and by default should be current year and Klingon respectively. The function should also verify that the language exists
+-- in the 'language' table. Then, ensure that no such function has been created before; if so, replace it.
+
+
+-- I dont understand for what we need add 'Klingon' language if we dont have it in language table ... 
+-- if we insert no language we got raise exception and can not add to db 
+
```
Comment (geroget) [inline SQL_DDL_Func_Oleg_Arslanov_HW.sql:294]: because this is the requirement specified in the task.

File: SQL_DDL_Func_Oleg_Arslanov_HW.sql [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+-- check if film exists in the DB
+    if exists (select 1 from public.film where upper(title) = upper(movie_title)
+	) then
+		raise exception 'Film "%" already exists in the DB!', movie_title;
+    end if;
+-- check if language name in language table if not add
+	if not exists (select 1 from public.language where upper(name) = upper(movie_language_name))
+	then	
+		insert into public.language (name)
+		values ('klingon');
```
Comment (geroget) [inline SQL_DDL_Func_Oleg_Arslanov_HW.sql:314]: You've hardcoded it!? And if I need to add a new film in Polish or Lithuanian, will your code allow me to do that?

**Review by geroget — APPROVED** [review]

**Comment by geroget** [comment]
Hello, please next time write comments in English

File: SQL_DDL_Func_Oleg_Arslanov_HW.sql (hunk 2 above)
Comment (olegarslanov) [reply SQL_DDL_Func_Oleg_Arslanov_HW.sql:314]: ok I wrote in english

**Pull request merged**